### DISCUSSION TITLE
Control DC motor from artisan

### DISCUSSION THREAD

**Discussion by xDESCAx** [artisan-roaster-scope/artisan#1750] [Q&A]
Hi all,

I am trying to control the speed of a motor from Artisan, I am trying to use the following phidgets:
- HUB0001_0
- TMP1101_1
- DCC1100_0 and DC motor
At the moment I can read the BT and ET temperatures but I don't know how to configure the motor, I want to control whit slider, is this possible?, I don't know much about electronics, anyone with ideas?

**Comment by MAKOMO** [comment]
The DCC1100 is not in the [list of supported devices](https://artisan-scope.org/devices/phidgets/) and I even couldn't find such a device on the [Phidget pages](https://phidgets.com/?view=search&q=DC1100).

**Reply by xDESCAx** [reply]
Hi Makomo, thanks for your answer, this is the phidget page for this dc controller: 
https://www.phidgets.com/?prodid=1013

And the motor im using is: 
https://www.phidgets.com/?prodid=1276

**Comment by MAKOMO** [comment]
As said, the DCC1100 is not in the list of supported devices. I don't understand why people purchase hardware that is not supported and expect it to work.

It is using a slightly different API then the DC motor controllers supported by Artisan as it is a brushless DC motor controller.  I just added support for it to the current [Continuous Builds](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) such that this Phidget should at least connect. 

I don't know anything about motor controllers and I cannot test this one for a lack of hardware. However, Phidget provides very detailed documentation for all of their products. Artisan is just implementing (parts of) their APIs. It seems you should be able to control the speed of it using the IO command `vel(0,{})` in a slider specification with a factor of `0.01` to have it range from 0 to 1 if you move the slider between 0 and 100.

<img width="1115" alt="Screenshot 2024-11-24 at 13 10 34" src="https://github.com/user-attachments/assets/e78d715e-7241-4ca9-846d-48927390d98d">
